enhancements/windows-containers/monitoring-windows-nodes.md

```diff
@@ -0,0 +1,144 @@
+  `openshift-windows-machine-config-operator` namespace. 
+* Creating a service and endpoints in `openshift-windows-machine-config-operator`
+  namespace that point to Windows exporter endpoint. WMCO will be using default
+  values to define metrics endpoint on Windows exporter which is equivalent to
+  `<internal-ip>:9182/metrics` for a given Windows node. The Endpoints object
+  created in the namespace will consist of subsets of endpoints from all the
+  Windows nodes.
+* Creating a service monitor in `openshift-windows-machine-config-operator` 
+  namespace for service created above.
+  
+To display node and workload graphs WMCO is responsible for the following:
+
+* Adding custom Prometheus rules in `openshift-windows-machine-config-operator` 
+  namespace. The custom recording rules will be created using Windows metrics
+  exposed by the Windows exporter and will have the same names as Linux
+  recording rules. This is to make use of same console queries as Linux.
+* Note that WMCO is unable to display workload graphs for the Windows Nodes. 
+  See [Risks and Mitigations](https://github.com/openshift/enhancements/blob/master/enhancements/windows-containers/monitoring-windows-nodes.md#risks-and-mitigations) 
+  for details.
+  
+### Risks and Mitigations
+
+The main risk with this proposal is renaming Windows metrics to display workload
+graphs. The pod metrics for linux come from cAdvisor. However, we do not get
+same metrics from cAdvisor for Windows nodes. This becomes a hindrance to 
+display pod graphs by creating custom recording rules to use same console 
+queries as Linux workloads.
+
+Mitigation
+* Use metrics exposed by the Windows exporter to display pod graphs. This also 
+  requires changes in console queries that support OS specific metrics. 
+
+## Design Details
+
+### Test Plan
+
+We plan to add e2e tests to ensure
+* Service, endpoints and Service Monitors are created as expected.
+* Prometheus is able to collect data from Windows exporter.
+* Custom Prometheus rules return Windows data. 
+
+### Graduation Criteria
+
+This enhancement will start as GA
+
+### Upgrade / Downgrade Strategy
+
+* WMCO is responsible for upgrading [Windows exporter](https://github.com/prometheus-community/windows_exporter/tags) 
+  binary to the latest release. 
+
+## Implementation History
+
+v1: Initial Proposal
+
+### Drawbacks
+
+Running Windows exporter as a Windows Service instead of running as a DaemonSet
+pod makes it hard for the Prometheus to monitor. The limitation of not able
+to run Windows exporter on Windows nodes as a pod is because of lack of
+privileged containers on Windows nodes.
+ 
+## Alternatives
+
+An alternative approach to display monitoring graphs is to create a common
+interface for Windows and Linux recording rules that is managed by
+cluster monitoring operator. The OpenShift console will utilize a single query
+as a result of this common interface which returns a metric irrespective of
+underlying OS.
```

Comment:
This is not an alternative but needs to be called out as a future plan with more details. That needs to be captured in this enhancement also. So the Implementation section needs to have a current state and a future state.

Reply:
Extract details from [WINC-530](https://issues.redhat.com/browse/WINC-530) and [ODC-5046](https://issues.redhat.com/browse/ODC-5046). We need to clearly show that there is monitoring and console work.